test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,821 @@
+				}
+			}
+			if profile.Spec.WorkloadHints == nil {
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+					RealTime: pointer.Bool(true),
+				}
+			}
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reservedSet,
+				Isolated: &isolatedSet,
+			}
+			profile.Spec.NUMA = &performancev2.NUMA{
+				TopologyPolicy: &policy,
+			}
+			By("Updating Performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		})
+
+		It("[test_id:60545] Reject guaranteed pod requesting resources that cannot be satisfied by 2 numa nodes together", func() {
+			var mm1 MMPod
+			mm1.memory = "200Mi"
+			mm1.cpu = "2"
+			mm1.noOfhpgs = "24Mi"
+			mm1.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			guPod := true
+			testPod := mm1.createPodTemplate(profile, guPod, &workerRTNodes[0])
+			By("creating test pod")
+			err = testclient.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to created test pod")
+			err = pods.WaitForCondition(testPod, corev1.PodConditionType(corev1.PodFailed), corev1.ConditionFalse, 2*time.Minute)
+			err := checkPodEvent(testPod, "TopologyAffinityError")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed), "Test pod does not have QoS class of Guaranteed")
+			err = mm1.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove test pod")
+		})
+
+		It("[test_id:60694] Accept guaranteed pod requesting resources that can be satisfied by 2 numa nodes together", func() {
+			var mm2 MMPod
+			mm2.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			targetNode := &workerRTNodes[0]
+			mm2.memory = "200Mi"
+			mm2.cpu = fmt.Sprintf("%d", len(isolated)-2)
+			// no. of hugepages is 20 * 2 (numazones). 40Mi
+			// we are asking for 30Mi, so it needs 2 numazones combined to
+			// satisfythe requirement
+			mm2.noOfhpgs = "30Mi"
+			testPod := mm2.createPodTemplate(profile, true, targetNode)
+			// Initialize test pod, check if the pod uses both numa node  0 and 1
+			initializePod(testPod, "0-1", targetNode)
+			err = mm2.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove test pod")
+		})
+
+		It("[test_id:60695] Allow burstable pod with hugepages", func() {
+			var mm2 MMPod
+			mm2.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			targetNode := &workerRTNodes[0]
+			mm2.memory = "200Mi"
+			mm2.cpu = fmt.Sprintf("%d", len(isolated)-2)
+			mm2.noOfhpgs = "8Mi"
+			testPod := mm2.createPodTemplate(profile, false, targetNode)
+			// Initialize test pod, check if the pod uses both numa node  0 and 1
+			initializePod(testPod, "0-1", targetNode)
```

Comment:
same here WRT initializePod(). it may result in duplicates but OTOH it exposes better the reasons for possible failures of the tests. Let me know what you think.

Reply:
@shajmakh Thanks you are right, i have addressed this in the latest commits